akka-remote/src/main/scala/akka/remote/artery/tcp/SSLEngineProvider.scala

```diff
@@ -73,119 +62,66 @@ class ConfigSSLEngineProvider(protected val config: Config, protected val log: M
       system.settings.config.getConfig("akka.remote.artery.ssl.config-ssl-engine"),
       Logging.withMarker(system, classOf[ConfigSSLEngineProvider].getName))
 
+  // Settings to read keystores (private keys and trusted certificates)
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLKeyStore: String = config.getString("key-store")
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLTrustStore: String = config.getString("trust-store")
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLKeyStorePassword: String = config.getString("key-store-password")
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLKeyPassword: String = config.getString("key-password")
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLTrustStorePassword: String = config.getString("trust-store-password")
-  val SSLEnabledAlgorithms: Set[String] = immutableSeq(config.getStringList("enabled-algorithms")).to(Set)
-  val SSLProtocol: String = config.getString("protocol")
+
+  // Set up RNG for the SSLEngine
+  @deprecated("Use SecureRandomFactory directly.", "2.6.6")
   val SSLRandomNumberGenerator: String = config.getString("random-number-generator")
-  val SSLRequireMutualAuthentication: Boolean = config.getBoolean("require-mutual-authentication")
-  val HostnameVerification: Boolean = config.getBoolean("hostname-verification")
```

Comment:
Instead of a list of values parsed from config in SSLEngineProvider, each piece (trust manager provider, secure random factory,...) reads the settings they need.